### ISSUE TITLE
[RyuJIT/arm32] Assertion failed 'tree->gtHasReg()'

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#11850]
At codegenlinear.cpp Line: 1158  

Example:
```
Assert failure(PID 12152 [0x00002f78], Thread: 13492 [0x34b4]): Assertion failed 'tree->gtHasReg()' in 'AutoGen.Program:Run():int:this' (IL size 80)

    File: c:\gh\coreclr\src\jit\codegenlinear.cpp Line: 1158
    Image: c:\brucefo\tests\Windows_NT.arm.Checked\Tests\Core_Root\CoreRun.exe
```

Tests with this assert:
```
JIT\jit64\regress\vsw\543229\test\test.cmd
```

**Comment by sjsinju** [comment]
@BruceForstall Please assign this to me.

**Comment by hseok-oh** [comment]
### Before morphing
```
               [000049] --C-G--------             /--*  callv ind int    AutoGen.Program.Func1
               [000003] ------------- this in r0  |  +--*  lclVar    ref    V00 this
               [000056] ------------- arg1        |  +--*  obj(2)    struct
               [000055] L------------             |  |  \--*  addr      byref
               [000017] -------------             |  |     \--*  lclVar    struct(AX)(P) V03 tmp0
                                                  |  |     \--*    byte   V03.f0 (offs=0x00) -> V09 tmp6
                                                  |  |     \--*    byte   V03.f1 (offs=0x01) -> V10 tmp7
               [000043] ------------- arg2        |  +--*  lclVar    long   V05 tmp2
               [000047] ------------- arg3        |  +--*  lclVar    int    V06 tmp3
               [000051] ------------- arg4        |  \--*  obj(16)   struct
               [000050] L------------             |     \--*  addr      byref
               [000048] -------------             |        \--*  lclVar    struct(AX)(P) V04 tmp1
                                                  |        \--*    long   V04.f0 (offs=0x00) -> V12 tmp9
                                                  |        \--*    long   V04.f1 (offs=0x08) -> V13 tmp10
               [000060] -AC-G--------             *  =         int
               [000058] U------N-----             \--*  lclFld    int    V07 tmp4         [+0]
                                                  \--*    byte   V07.f0 (offs=0x00) -> V14 tmp11
                                                  \--*    byte   V07.f1 (offs=0x01) -> V15 tmp12
```

### Before
```
fgMorphTree BB01, stmt 7 (after)
               [000049] --C-G+-------             /--*  callv ind int    AutoGen.Program.Func1
               [000003] -----+-------             |  |  /--*  lclVar    ref    V00 this
               [000213] -A---------L- this SETUP  |  +--*  =         ref
               [000212] D------N-----             |  |  \--*  lclVar    ref    V26 tmp23
               [000209] ----G--N-----             |  |     /--*  lclVar    byte  (AX) V10 tmp7
               [000210] -A--G--------             |  |  /--*  =         byte    indir assign of V24:ud:0->0
               [000208] *------N-----             |  |  |  \--*  indir     byte
               [000206] -------------             |  |  |     |  /--*  const     int    1 Fseq[f1]
               [000207] -------------             |  |  |     \--*  +         byref
               [000205] -------------             |  |  |        \--*  lclVar    byref  V25 tmp22
               [000211] -A--G+-----L- arg1 SETUP  |  +--*  comma     void
               [000202] ----G--N-----             |  |  |     /--*  lclVar    byte  (AX) V09 tmp6
               [000203] -A--G--------             |  |  |  /--*  =         byte    indir assign of V24:ud:0->0
               [000201] *------N-----             |  |  |  |  \--*  indir     byte
               [000199] -------------             |  |  |  |     |  /--*  const     int    0 Fseq[f0]
               [000200] -------------             |  |  |  |     \--*  +         byref
               [000198] -------------             |  |  |  |        \--*  lclVar    byref  V25 tmp22
               [000204] -A--G--------             |  |  \--*  comma     void
               [000194] L------------             |  |     |  /--*  addr      byref
               [000195] -------N-----             |  |     |  |  \--*  lclVar    struct(AX) V24 tmp21
               [000197] -A-----------             |  |     \--*  =         byref
               [000196] D------N-----             |  |        \--*  lclVar    byref  V25 tmp22
               [000047] -----+------- arg3 out+00 |  +--*  lclVar    int    V06 tmp3
     (  9,  7) [000051] ----G-------- arg4 out+08 |  +--*  obj(16)   struct
     (  3,  3) [000050] L------------             |  |  \--*  addr      byref
     (  3,  2) [000048] ----G--N-----             |  |     \--*  lclVar    struct(AX)(P) V04 tmp1
                                                  |  |     \--*    long   V04.f0 (offs=0x00) -> V12 tmp9
                                                  |  |     \--*    long   V04.f1 (offs=0x08) -> V13 tmp10
               [000214] ------------- this in r0  |  +--*  lclVar    ref    V26 tmp23
               [000223] ------------- arg1 in r1  |  +--*  <fldList> void   int at offset 0
               [000222] -------------             |  |  \--*  lclFld    int    V24 tmp21        [+0]
               [000043] -----+------- arg2 r2,r3  |  \--*  lclVar    long   V05 tmp2
               [000060] -AC-G+-------             *  =         int
               [000058] U----+-N-----             \--*  lclFld    int    V07 tmp4         [+0]
                                                  \--*    byte   V07.f0 (offs=0x00) -> V14 tmp11
                                                  \--*    byte   V07.f1 (offs=0x01) -> V15 tmp12
```

### After
```
fgMorphTree BB01, stmt 7 (after)
               [000049] --C-G+-------             /--*  callv ind int    AutoGen.Program.Func1
               [000003] -----+-------             |  |  /--*  lclVar    ref    V00 this
               [000213] -A---------L- this SETUP  |  +--*  =         ref
               [000212] D------N-----             |  |  \--*  lclVar    ref    V26 tmp23
               [000209] ----G--N-----             |  |     /--*  lclVar    byte  (AX) V10 tmp7
               [000210] -A--G--------             |  |  /--*  =         byte    indir assign of V24:ud:0->0
               [000208] *------N-----             |  |  |  \--*  indir     byte
               [000206] -------------             |  |  |     |  /--*  const     int    1 Fseq[f1]
               [000207] -------------             |  |  |     \--*  +         byref
               [000205] -------------             |  |  |        \--*  lclVar    byref  V25 tmp22
               [000211] -A--G+-----L- arg1 SETUP  |  +--*  comma     void
               [000202] ----G--N-----             |  |  |     /--*  lclVar    byte  (AX) V09 tmp6
               [000203] -A--G--------             |  |  |  /--*  =         byte    indir assign of V24:ud:0->0
               [000201] *------N-----             |  |  |  |  \--*  indir     byte
               [000199] -------------             |  |  |  |     |  /--*  const     int    0 Fseq[f0]
               [000200] -------------             |  |  |  |     \--*  +         byref
               [000198] -------------             |  |  |  |        \--*  lclVar    byref  V25 tmp22
               [000204] -A--G--------             |  |  \--*  comma     void
               [000194] L------------             |  |     |  /--*  addr      byref
               [000195] -------N-----             |  |     |  |  \--*  lclVar    struct(AX) V24 tmp21
               [000197] -A-----------             |  |     \--*  =         byref
               [000196] D------N-----             |  |        \--*  lclVar    byref  V25 tmp22
               [000047] -----+------- arg3 out+00 |  +--*  lclVar    int    V06 tmp3
     (  9,  7) [000051] ----G-------- arg4 out+08 |  +--*  obj(16)   struct
     (  3,  3) [000050] L------------             |  |  \--*  addr      byref
     (  3,  2) [000048] ----G--N-----             |  |     \--*  lclVar    struct(AX)(P) V04 tmp1
                                                  |  |     \--*    long   V04.f0 (offs=0x00) -> V12 tmp9
                                                  |  |     \--*    long   V04.f1 (offs=0x08) -> V13 tmp10
               [000214] ------------- this in r0  |  +--*  lclVar    ref    V26 tmp23
               [000216] ------------- arg1 in r1  |  +--*  lclFld    short  V24 tmp21        [+0]
               [000043] -----+------- arg2 r2,r3  |  \--*  lclVar    long   V05 tmp2
               [000060] -AC-G+-------             *  =         int
               [000058] U----+-N-----             \--*  lclFld    int    V07 tmp4         [+0]
                                                  \--*    byte   V07.f0 (offs=0x00) -> V14 tmp11
                                                  \--*    byte   V07.f1 (offs=0x01) -> V15 tmp12
```

**Comment by sjsinju** [comment]
@BruceForstall Resolved by #12079, Close this please.
